urllib3/response.py

```diff
@@ -178,6 +188,8 @@ def read(self, amt=None, decode_content=None, cache_content=False):
             after having ``.read()`` the file object. (Overridden if ``amt`` is
             set.)
         """
+        if self.chunked:
+            return
```

Comment:
TBH I'm not really sure how to approach this: since the response is chunked and will be continually growing (which may take hours), read may block for a long time. Is this something we really want?

But as I'm thinking about it: `read` may be the blocking call, and `read_chunked` the non-blocking asynchronous call. How about that?